### DISCUSSION TITLE
https://notes.alexkehayias.com/capture-output-of-chatgpt-shell-in-org-mode/

### DISCUSSION THREAD

**Discussion by giscus** [alexkehayias/note-comments#6] [Announcements]
# https://notes.alexkehayias.com/capture-output-of-chatgpt-shell-in-org-mode/

I’m messing around with ChatGPT in Emacs and I want to capture the results of a session in org-mode.

https://notes.alexkehayias.com/capture-output-of-chatgpt-shell-in-org-mode/

**Comment by lesliesrussell** [comment]
I couldn't get your version to work so I tried tweaking it a little. It got out of hand a bit:
```lisp
(defun find-closest-buffer-match (partial-name)
  "Find the closest matching buffer name to the given partial name."
  (let* ((all-buffer-names (mapcar #'buffer-name (buffer-list)))
         (matching-buffers (seq-filter (lambda (name) (string-match-p partial-name name)) all-buffer-names))
         (closest-buffer (car matching-buffers)))
    closest-buffer))

(setq partial-name "chat")
(setq closest-match-buffer (find-closest-buffer-match partial-name))

(if closest-match-buffer
    (let ((buf (get-buffer-create closest-match-buffer)))
      (with-current-buffer buf
	(buffer-substring-no-properties (point-min) (point-max))))
  (message "No matching buffer found"))
```

**Reply by alexkehayias** [reply]
The author of `chatgpt-shell` added an org-babel way to do this. https://github.com/xenodium/chatgpt-shell#chatgpt-org-babel

Take a look to see if that does what you are looking to do!
